src/erlcloud_cloudtrail.erl

```diff
@@ -172,9 +173,7 @@ update_trail(Trail, S3BucketName, S3KeyPrefix, SnsTopicName, IncludeGlobalServic
 
 % Json parameter must be a list of binary key/value tuples.
 ct_request(Operation, [], Config) ->
-    #aws_config{cloudtrail_scheme = Scheme, 
-                cloudtrail_host = Host} = Config,
-    request_impl(post, Scheme, Host, port_spec(Config), "/", Operation, [], "{}", Config);
+    ct_request(Operation, "{}", Config);
```

Comment:
This clause defined for empty `Body` param was doing exactly the same thing as the clause with non-empty `Body` except it was passing `Body` as a string instead of binary. The functions this `Body` param was getting passed to are expecting a binary so dialyzer was complaining. Seems like the purpose of this clause was to define a special case for empty body and instead of calling `jsx:encode([])` it just has the result of this call hard coded. The problem is: the result of `jsx:encode([])` is `<<"[]">>` and not `"{}"`. I suppose the correct solution would be to either remove the clause completely or have this clause same as it was but with `"{}"` replaced with `<<"[]">>`